### PR TITLE
Architecture diagram added

### LINKED ISSUE #877: Extend Architecture diagram
The [diagram](https://www.k8gb.io/docs/components.html) contains one cluster connected to one edgeDNS. What is missing in the diagram is a DNS query, where the k8gb controller queries the external target(s) and updates its own dnsEndpoints.

Perhaps it would be useful to add a new image explaining this.

### PR DESCRIPTION
**Pull request by kuritka**
closes #877

Signed-off-by: kuritka <kuritka@gmail.com>

### CODE REVIEW

File: docs/components.md [hunk 1]
```diff
@@ -2,3 +2,6 @@
 
 [![K8GB Internal Components](/docs/images/k8gb-components.svg)](/docs/images/k8gb-components.svg)
 *The flow between internal k8gb components including scenario of Route53 integration.*
+
+[![K8GB Cluster Collaboration](/docs/images/k8gb-collaboration.svg)](/docs/images/k8gb-collaboration.svg)
```
Comment (somaritane) [inline docs/components.md:6]: @kuritka not sure if "cluster collaboration" is the right term here, maybe "multi-cluster interoperability"?

File: docs/images/k8gb-collaboration.svg [hunk 2]
```diff
@@ -0,0 +1,4 @@
...
... 136px; margin-left: 1296px;"><div data-drawio-colors="color: rgb(0, 0, 0); " style="box-sizing: border-box; font-size: 0px; text-align: left;"><div style="display: inline-block; font-size: 12px; font-family: Helvetica; color: rgb(0, 0, 0); line-height: 1.2; pointer-events: all; white-space: normal; overflow-wrap: normal;"><i>* The following diagrams describe the collaboration between k8gb regions (clusters). The diagram is only for easy understanding of the collaboration between clusters, not step by step what exactly is happening.  The runtime sets the steps in random order due to TTL, and the complete decentralization of all components.    </i></div></div></div></foreignObject><text x="1296" y="139" fill="rgb(0, 0, 0)" font-family="Helvetica" font-size="12px">* The following diagrams describe the collaboration between k8gb regions (clusters)...</text></switch></g></g><switch><g requiredFeatures="http://www.w3.org/TR/SVG11/feature#Extensibility"/><a transform="translate(0,-5)" xlink:href="https://www.diagrams.net/doc/faq/svg-export-text-problems" target="_blank"><text text-anchor="middle" font-size="10px" x="50%" y="100%">Text is not SVG - cannot display</text></a></switch></svg>
```
Comment (somaritane) [inline docs/images/k8gb-collaboration.svg:4]: The diagram description under the title still reads "Internal k8gb components and relation between them".
Should it still be the case?
image

**Review by somaritane — CHANGES_REQUESTED** [review]
@kuritka thank you for arch diagram update, looks good with a few comments from me.

**Comment by kuritka** [comment]
@somaritane , @ytsarev  all required changes amended. thanks!

**Review by somaritane — APPROVED** [review]
@kuritka lgtm 👍

**Pull request merged**